pkg/client/client.go

```diff
@@ -197,11 +210,16 @@ func newClient(config *rest.Config, options Options) (*client, error) {
 		},
 		metadataClient: metadataClient{
 			client:     rawMetaClient,
-			restMapper: options.Mapper,
+			restMapper: options.MapperWithContext,
 		},
 		scheme: options.Scheme,
 		mapper: options.Mapper,
 	}
+	mapperCache, err := lru.New[logicalcluster.Name, meta.RESTMapper](1000)
```

Comment:
Would it make sense 1000 to create a global variable? If somebody wants to tune it, its possible this way?

Reply:
Addressed.